### ISSUE TITLE
Omnidirectional Keyboard/Mouse Sharing (Support for USB KVM)

### ISSUE THREAD

**Issue by hicklemon** [symless/synergy-core#6991]
TL;DR: 2 or more computers, use the keyboard/mouse on ANY of the machines to control the cursor/input.

I realize this feature request is an edge case, and probably tons of work, or impossible. But I wanted to get this out there in case anyone else has tried this before. And I also realize there is nearly no other use case for this feature.

I have been using a USB KVM to manually switch my mouse, keyboard, webcam, audio interface, etc. between a Windows and a Linux machine.

Currently, I've set up Synergy on the Windows box as the server and Synergy on the Linux box as the client. This works. If my KVM is set to my Windows machine, I can interact with my Linux machine... as long as it's in a desktop environment and Synergy is actually running.

Sometimes, though, I need to switch to a TTY on the Linux box. Or sometimes, I need to get into the BIOS on the Windows box. And in those situations, I use my USB KVM to switch. That's fine. I have to reach over and click a button. But I'm lazy (which is why I use Synergy).

In my laziness, I want to have my keyboard and mouse hooked up to either my Linux box or my Windows box at any given time depending on whether I need to be in a non-graphical TTY, or in the BIOS, or whatever.

I have tried mixing Synergy and barrier (https://github.com/debauchee/barrier) to achieve this, configuring Synergy as the server on the Windows box, and Barrier as the server on the Linux box, but when they're both running and they connect with their counterparts (and they do connect!), they immediately confuse each other about where the mouse is, what has keyboard focus, and which machine has hardware control over it from which side. Essentially, the cursor disappears, and keyboard input does not work (except if I KVM over to the Linux box and Ctrl-Alt-F2 i _can_ switch TTY).

So the solution would have to be in software on both sides. A bidirectional server setup. Similar to primary-primary database servers where it doesn't matter which one you send a write to, the query just "runs".

**Comment by hicklemon** [comment]
In hindsight this wouldn't work with a BIOS or TTY.

bidirectional or omnidirectional synergy in a desktop environment with any m/kb from any machine sounds cool and all and should work but there's probably some ownership of control issue that can't be beaten.

Closing

**Comment by MrJacquers** [comment]
MouseWithoutBorders has bi-directional support. It doesn't run under Linux though, although it may be possible via Wine.

**Comment by aaronth07** [comment]
Reopened. Mouse without borders does not have Mac support (AFAIK).
